category/category.go

```diff
@@ -0,0 +1,165 @@
+package category
+
+import (
+	"context"
+
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/gormsupport"
+	"github.com/almighty/almighty-core/log"
+	"github.com/jinzhu/gorm"
+	uuid "github.com/satori/go.uuid"
+)
+
+// Defines "type" string to be used while validating jsonapi spec based payload
+const (
+	APIStringTypeCategory = "categories"
```

Comment:
I think `APIStringType` is enough, since it will be referenced by `category.APIStringType`.